### PR TITLE
Use SavedStateHandle in view models

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
Pass [`SavedStateHandle`](https://developer.android.com/reference/androidx/lifecycle/SavedStateHandle) instead of `Bundle` to the view models. Values provided via an Intent's extras or a fragment's arguments are automatically passed to the state handle in the default view model factory implementation, which eliminates the need to implement a view model factory separately.

### CODE REVIEW

**Review by dbrant — CHANGES_REQUESTED** [review]
Thanks -- this looks like it will be a worthwhile change. However, a change of this size requires a _much better description_.
The word "simplify" is not meaningful enough. There is a considerable reduction in code, but _why_ is there a reduction in code?

Is it because `SavedStateHandle` automatically provides access to the Activity's intent extras and/or the Fragment's arguments? If so, this should be very clearly specified in your description. I'm not even seeing this described in Google's own documentation of `SavedStateHandle`, which worries me a bit. Can you point to where in the documentation does it endorse using `SavedStateHandle` in this way?

**Review by machado001 — CHANGES_REQUESTED** [review]
Minor comments =)

File: app/src/main/java/org/wikipedia/page/edithistory/EditHistoryListViewModel.kt [hunk 1]
```diff
@@ -1,33 +1,33 @@
 package org.wikipedia.page.edithistory
 
-import android.os.Bundle
 import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.SavedStateHandle
 import androidx.lifecycle.ViewModel
-import androidx.lifecycle.ViewModelProvider
 import androidx.lifecycle.viewModelScope
 import androidx.paging.*
-import kotlinx.coroutines.*
```
[the remark this answers is not in the document]
Comment (Isira-Seneviratne) [reply app/src/main/java/org/wikipedia/page/edithistory/EditHistoryListViewModel.kt:9]: That wasn't intentional, thanks for pointing it out.

**Review by machado001 — APPROVED** [review]

**Comment by trietbui85** [comment]
This PR gets bigger and bigger. Should we separate it to multiple smaller PRs?

**Pull request merged**